sycl/source/detail/context_impl.cpp

```diff
@@ -46,12 +46,14 @@ context_impl::context_impl(const vector_class<cl::sycl::device> Devices,
   const auto Backend = getPlugin().getBackend();
   if (Backend == backend::cuda) {
 #if USE_PI_CUDA
-    const pi_context_properties props[] = {
+    const bool UseCUDAPrimaryContext =
+        MPropList.has_property<property::context::cuda::use_primary_context>();
+    const pi_context_properties Props[] = {
         static_cast<pi_context_properties>(PI_CONTEXT_PROPERTIES_CUDA_PRIMARY),
         static_cast<pi_context_properties>(UseCUDAPrimaryContext), 0};
 
     getPlugin().call<PiApiKind::piContextCreate>(
-        props, DeviceIds.size(), DeviceIds.data(), nullptr, nullptr, &MContext);
+        Props, DeviceIds.size(), DeviceIds.data(), nullptr, nullptr, &MContext);
 #else
     cl::sycl::detail::pi::die("CUDA support was not enabled at compilation time");
```

Comment:
Not related to this patch, but this message is misleading. If I saw this in my program, I would have started looking for dpcpp compiler options to enable CUDA.